bench/workgen/workgen.cxx

```diff
@@ -882,7 +882,8 @@ int ThreadRunner::op_run(Operation *op) {
 
             _in_transaction = true;
         }
-        if (op->is_table_op()) {
+        // Call cursor operations only when transaction != NULL or _in_transaction is true.
+        if (op->is_table_op() && _in_transaction) {
```

Comment:
@raviprakashgiri29, I don't think this is right.  Doesn't this say that if you're not using transactions, no operations occur?  We want to support workloads that have no transactions.

Reply:
I suppose we could have logic that said if some transactions used one isolation level and other transactions used another one, or didn't explicitly use a transaction, then we would issue some sort of error within workgen.  But that's a lot more work than we want to do to catch a script error.  I think just fixing the script is the right solution.

Reply:
@ddanderson : I don't see any issue in the script. UPDATE operation in the script is wrapped under a transaction (`txn`) with snapshot isolation level but the code flow hits the UPDATE `cursor->update()` without snapshot isolation level. 
I think some issue in the workgen.cxx file in this case wherein the code calls update operation `cursor->update()` without setting `begin_transaction isolation level` to snapshot and by default the snapshot level is `read-committed`.

Reply:
Sorry if I wasn't clear.  I think your changes to `example_prepare.py` are fine.  I don't think the changes to `workgen.cxx` are correct, as that will have the effect of having scripts that don't use transactions (like `example_simple.py`) not do any work (no inserts will actually be done).  To make sure I applied just the `example_prepare.py` changes, and running that example worked.  Let's check in on this next week.

Reply:
@ddanderson : Thank you for correcting me. I understand your concern here and I agree that checking for `_in_transaction` will block workloads running without any transaction. 

I will revert the change and update the PR to pass the transaction begin_config i.e, into `conn->open_session()`. 
ex: workloads running without transaction will pass the NULL config string and it is backward compatible.